### DISCUSSION TITLE
How to achieve scheduler.PauseAll(); and  scheduler.ResumeAll(); or standby in dotnet core quartz 3.x

### DISCUSSION THREAD

**Discussion by VidhyaSanjeevi** [quartznet/quartznet#1722] [Q&A]
**Describe the bug**

Not a bug, its an how to question - how to make pause all or resume all or standby for all job or triggers when using quartz 3.x in dotnet core 3.1 or dotnet 6?

**Version used**

3.4

**To Reproduce**

Follow [this ](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/aspnet-core-integration.html#installation) documentation and implement quartz scheduler job with triggers and then we need to pause all or standby for all jobs or triggers.

**Expected behavior**

A way to make it standby.

**Comment by lahma** [comment]
Did you already explore the IScheduler API?

**Reply by VidhyaSanjeevi** [reply]
the thing is how can i get the scheduler object here?

tried below - it doesn't makes it to standby - i think its creating new object instead of getting the default scheduler created in above code.
```

var scheduler = await StdSchedulerFactory.GetDefaultScheduler();
await scheduler.Standby();
await scheduler.PauseTrigger(new TriggerKey(workerTriggerKeyName));
```

**Reply by lahma** [reply]
You should build the scheduler factory only once for given configuration and then you can call `GetScheduler()` multiple times on it and always get the same instance. 

Also check the MS DI documentation and with that just resolve the `ISchedulerFactory`.

**Reply by VidhyaSanjeevi** [reply]
I have created and mapped the schedulerfactory by this:

```

                services.AddSingleton<ISchedulerFactory>(con => GetSchedulerFactory());
                services.AddSingleton<IScheduler>(con => GetScheduler());



        private static IScheduler GetScheduler()
        {
            if (scheduler == null)
            {
                // Create a Quartz.NET scheduler
                scheduler = GetSchedulerFactory().GetScheduler().Result;
            }
            return scheduler;
        }

        private static ISchedulerFactory GetSchedulerFactory()
        {
            if (schedulerFactory == null)
            {
                // Create a Quartz.NET scheduler
                schedulerFactory = new StdSchedulerFactory();
                //scheduler = schedulerFactory.GetScheduler().Result;
            }
            return schedulerFactory;
        }
```

But after this the trigger is not triggered and job doesn't runs.

**Reply by lahma** [reply]
https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/microsoft-di-integration.html

**Reply by NxSoftware** [reply]
I have basically the same problem, I want to create a job & trigger that is paused by default then later resumed after some domain-specific condition is met.

We can't get an instance of `ISchedulerFactory` at **startup** as we are still configuring Quartz via DI at that point.

The only way I've found to make this work is to inject `ISchedulerFactory` into in the `Startup.Configure` method...

```csharp
var scheduler = schedulerFactory.GetScheduler().Result;
scheduler.PauseJobs(GroupMatcher<JobKey>.GroupEquals("GroupToBePausedByDefault"));
```

The use of `.Result` on the returned `Task` feels a bit hacky, though I'm not sure if it's OK in this instance.
